Dynatrace/dynatrace-operator
# Introduce DynaKube v1beta2

## Description



Same as https://github.com/Dynatrace/dynatrace-operator/pull/3092, but rebased to `main`.
As the original PR was sooo old and had multiple merges back from main, the state of it was rather "cursed".

⚠️  Pls review it with care, as its maaany files.

## How can this be tested?
Create a `DynaKube` with version `v1beta1`, it should be converted without any issues/
Create a `DynaKube` with version `v1beta2`, it should be used as expected.

(I didn't have time to test it manually yet, but wanted to open the PR so people have more time to look at it 😬 )
